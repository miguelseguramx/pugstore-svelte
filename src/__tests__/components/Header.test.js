import { render } from '@testing-library/svelte'
import Header from '../../components/Header.svelte';

describe('Header component', () => {
  const { getByTestId } = render(Header)
  const header = getByTestId('Header')

  it('It renders the header!', () => {
    expect(header).toBeInTheDocument()
  })

  it('Check the logo', () => {
    const logo = header.getElementsByTagName('h1')[0]
    expect(logo).toHaveTextContent('Pugstagram')
  })
  
  it('Check the likes icon', () => {
    const likes = header.getElementsByTagName('i')[0]
    expect(likes).toHaveClass('fas fa-heart')
  })
  
  it('Check the followers icon', () => {
    const followers = header.getElementsByTagName('i')[1]
    expect(followers).toHaveClass('fas fa-user-alt')
  })
})

describe('Header SnapShot', () => {
  test('Check the Snapshot of the Header', () => {
    const { getByTestId } = render(Header)
    const header = getByTestId('Header')
    expect(JSON.stringify(header)).toMatchSnapshot()
  })
})