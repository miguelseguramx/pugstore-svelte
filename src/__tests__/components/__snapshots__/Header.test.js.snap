// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header SnapShot Check the Snapshot of the Header 1`] = `"{\\"__svelte_meta\\":{\\"loc\\":{\\"file\\":\\"Header.svelte\\",\\"line\\":62,\\"column\\":0,\\"char\\":1245}}}"`;
